test/integration/http_protocol_integration.cc

```diff
@@ -50,15 +50,17 @@ std::vector<HttpProtocolTestParams> HttpProtocolIntegrationTest::getProtocolTest
 #else
         use_header_validator_values.push_back(false);
 #endif
-        for (Http1ParserImpl http1_implementation : http1_implementations) {
-          for (Http2Impl http2_implementation : http2_implementations) {
-            for (bool defer_processing : http2_bool_values) {
-              for (bool deprecate_callback_visitor : http2_bool_values) {
-                for (bool use_header_validator : use_header_validator_values) {
-                  ret.push_back(HttpProtocolTestParams{
-                      ip_version, downstream_protocol, upstream_protocol, http1_implementation,
-                      http2_implementation, defer_processing, use_header_validator,
-                      deprecate_callback_visitor});
+        for (const bool tunneling_with_upstream_filters : {false, true}) {
```

Comment:
Please find a compromise between a reasonable amount of coverage, and doubling the number of permutations of the tests. Recent failures were partially caused by the tests taking too long due to too many permutations.